cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1351,14 +1355,26 @@ func (p *BicepProvider) createOutputParameters(
 	return outputParams
 }
 
+type bicepModule struct {
```

Comment:
I wonder if this structure buys us much.  fileType can be inferred from the extension by construction.  I think things would be simpler if we just did things like `filepath.Ext(<module-path>) == ".bicep" instead of having `bicepModule` and `bicepModuleType` which starts to feel like a lot of ceremony.